pkg/cmd/registry/artifact/crud/list/list.go

```diff
@@ -0,0 +1,210 @@
+			}
+
+			if len(args) > 0 {
+				opts.group = args[0]
+			}
+
+			if opts.page < 1 || opts.limit < 1 {
+				return errors.New("page and limit values should be bigger than 1")
+			}
+
+			if opts.registryID != "" {
+				return runList(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return errors.New("No service Registry selected. Use rhoas registry use to select your registry")
+			}
+
+			opts.registryID = fmt.Sprint(cfg.Services.ServiceRegistry.InstanceID)
+
+			return runList(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.group, "group", "g", "", "Artifact group")
+
+	cmd.Flags().Int32VarP(&opts.page, "page", "", 1, "Page number")
+	cmd.Flags().Int32VarP(&opts.limit, "limit", "", 100, "Page limit")
+
+	cmd.Flags().StringVarP(&opts.registryID, "instance-id", "", "", "Id of the registry to be used. By default uses currently selected registry")
+	cmd.Flags().StringVarP(&opts.outputFormat, "output", "o", "", "Output format (json, yaml, yml, table)")
+
+	flagutil.EnableOutputFlagCompletion(cmd)
+
+	return cmd
+}
+
+func runList(opts *Options) error {
+	logger, err := opts.Logger()
+	if err != nil {
+		return err
+	}
+
+	if opts.group == "" {
+		logger.Info("Group was not specified. Using " + util.DefaultArtifactGroup + " artifacts group.")
+		opts.group = util.DefaultArtifactGroup
+	}
+
+	connection, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api := connection.API()
+
+	a, _, err := api.ServiceRegistryInstance(opts.registryID)
+	if err != nil {
+		return err
+	}
+	request := a.ArtifactsApi.ListArtifactsInGroup(context.Background(), opts.group)
+
+	request = request.Offset((opts.page - 1) * opts.limit)
+	request = request.Limit(opts.limit)
+	request = request.Orderby(registryinstanceclient.SORTBY_CREATED_ON)
+	request = request.Order(registryinstanceclient.SORTORDER_ASC)
+
+	response, _, err := request.Execute()
+	if err != nil {
+		return registryinstanceerror.TransformError(err)
+	}
+
+	totalCount := opts.page * opts.limit
+	if len(response.Artifacts) != 0 && response.GetCount() < totalCount {
+		logger.Info("Provided limit and page arguments are larger than total count of elements on the server", response.GetCount())
+		return nil
+	}
+
+	if len(response.Artifacts) == 0 && opts.outputFormat == "" {
+		logger.Info("No artifacts available for " + opts.group + " group and registry id " + opts.registryID)
+		return nil
+	}
+
+	switch opts.outputFormat {
+	case "json":
+		data, _ := json.Marshal(response)
+		_ = dump.JSON(opts.IO.Out, data)
+	case "yaml", "yml":
+		data, _ := yaml.Marshal(response)
+		_ = dump.YAML(opts.IO.Out, data)
+	default:
+		rows := mapResponseItemsToRows(&response.Artifacts)
+		dump.Table(opts.IO.Out, rows)
+		logger.Info("")
+	}
+
+	return nil
+}
+
+func mapResponseItemsToRows(artifacts *[]registryinstanceclient.SearchedArtifact) []artifactRow {
```

Comment:
```suggestion
func mapResponseItemsToRows(artifacts []registryinstanceclient.SearchedArtifact) []artifactRow {
```

A slice does not store any data, it just describes a section of an underlying array. Therefore a pointer is not needed.